mesonbuild/options.py

```diff
@@ -445,13 +431,12 @@ def extend_value(self, value: T.Union[str, T.List[str]]) -> None:
         self.set_value(self.value + new)
 
 
+@dataclasses.dataclass
 class UserFeatureOption(UserComboOption):
-    static_choices = ['enabled', 'disabled', 'auto']
 
-    def __init__(self, name: str, description: str, value: T.Any, yielding: bool = DEFAULT_YIELDING,
-                 deprecated: T.Union[bool, str, T.Dict[str, str], T.List[str]] = False):
-        super().__init__(name, description, self.static_choices, value, yielding, deprecated)
-        self.name: T.Optional[str] = None  # TODO: Refactor options to all store their name
+    choices: T.List[str] = dataclasses.field(
+        # Ensure we get a copy with the lambda
+        default_factory=lambda: ['enabled', 'disabled', 'auto'], init=False)
```

Comment:
Why not simply generate is in `__post_init__` instead?

Reply:
I dislike using `__post_init__`, tbh. It feels like a code smell to me. IMO, it's also distraction because it moves from "here are all of the fields for my class in one place" to "fields for my class are declared in multiple places

Reply:
one way or other are ok for me.